### ISSUE TITLE
`[MaybeNull]` does not work for parameters

### ISSUE THREAD

**Issue by CollinAlpert** [dotnet/roslyn#68609]
**Version Used**: 4.6.0-2.23152.6 (1314d090)

**Steps to Reproduce**:
Consider the following program:
```csharp
using System.Diagnostics.CodeAnalysis;

#nullable enable

void Run([MaybeNull] string s)
{
    Console.WriteLine(s.Length);
}
```

**Diagnostic Id**: CS8321

**Expected Behavior**: CS8321 gets raised in line 7.

**Actual Behavior**: No warning is emitted. I also tried using `[param: MaybeNull]`, but that made no difference.

The documentation for `MaybeNull` states "Specifies that an output may be null even if the corresponding type disallows it." and I am confused by the "output" phrasing, however the `AttributeTargets` also specify parameters, so I would expect it to work like in my example above.

**Comment by CyrusNajmabadi** [comment]
`MaybeNull` is a postcondition attribute.  it tells the compiler something about a value *after* something has happened (in this case, after `Run` has been called.  It is the caller that will learn something.

**Comment by CollinAlpert** [comment]
I see, thanks for the explanation. Does an attribute exist to achieve my use case?

**Comment by CyrusNajmabadi** [comment]
It's not exactly clear to me what semantics you want. But I imagine that you could just write `string? s` here too get it.
